### DISCUSSION TITLE
Converting pyrolysis model from FDS 6.7.7 to 6.9.0

### DISCUSSION THREAD

**Discussion by r-broek** [firemodels/fds#12709] [Q&A] [answered]
Dear developers,

At the moment, I am trying to make a vegetation model developed for FDS 6.7.7 work for FDS 6.9.0. In particular, the FDS user guide suggests that the pre-exponential factor $A_{ij}$ in the complex pyrolysis model should be adjusted for reactions of order $n_s \neq 1$ (footnote p. 96). Unfortunately, I do not understand the suggested substitution $A_{ij} = A_{ij}^{old} / \left( \rho_s(0)^{n_s-1}\right)$.

 I have compared the reaction rate equations given in the user guide of the respective versions (Eq. 11.7 for 6.7.7 and Eq. 9.2 for 6.9.0), and I think it would seem more suitable to have a substitution of $A_{ij} = A_{ij}^{old} /\left( \rho_s(0)^{n_s}\right)$. However, in this case, the substitution no longer reduces to $A_{ij} = A_{ij}^{old}$ for $n_s = 1$. This instead happens for $n_s = 0$.

Since I am working with a vegetation model similar to the NIST Douglas Fir validation model, which considers drying (first order), pyrolysis (first order) and char oxidation (zeroth order), I am confused about what to adjust. I have tried the suggested adjustment for the char oxidation: $A_{ij} = A_{ij}^{old} / \rho_s(0)^{0-1} = A_{ij}^{old} \times \rho_s(0)$. This did not significantly affect the simulation results. I have also looked at the committed changes to the vegetation model, but I could not deduce the substitution.

Furthermore, if I understand correctly $\rho_s$ is the composite density of the layer. Does this mean that $\rho_s(0) = (1-\mathrm{MC}) \cdot\rho_{veg} + \mathrm{MC}\cdot \rho_{H2O}$ for a vegetation particle and is thus dependent on the moisture content?

**Comment by mcgratta** [comment]
You are correct in thinking that if $n_s=1$, the value of $A_{ij}$ should not change. Your comment that the change did not significantly affect the simulation results does not mean that the change is wrong.

**Reply by r-broek** [reply]
Okay, that helps already, thank you! I would still appreciate it if someone could explain why the suggested substitution was chosen.

Regarding the simulation results, I have looked at the FDS release notes for any changes that could explain the change in simulation results between the versions. I concluded that it is likely caused by the change in the reaction rate equations, the change of the SURFACE_OXIDATION_MODEL, or both.  Since correctly changing $A_{ij}$ had no effect, do you agree that the difference could be due to the adjustment of the SURFACE_OXIDATION_MODEL? Or am I forgetting any other important alterations that may impact the results significantly?

**Comment by mcgratta** [comment, marked as answer]
The original FDS reaction rate expression differed from that used by other pyrolysis researchers. We decided to conform to the way it is commonly done by the community. That is, we removed the $rho_s(0)$ from the denominator. 

As far as your other questions -- yes, most things related to vegetation have changed in the last few versions. This means that previously published kinetic constants may not yield the same results in the latest version. My only recommendation to you is to get the raw TGA data for a particular species and derive the appropriate parameters yourself. Use the `TGA_ANALYSIS` feature in FDS to check the modeled TGA results directly against the experimental data. Do not rely on published parameters. There is no consistency among models, or versions of models, in regard to these parameters.

**Reply by r-broek** [reply]
The transformation from $Y$ to $\rho_s$ I understand, but I still do not see how they are equivalent through the substitution of $A_{ij}$. For $n_s=1$ the suggested change implies $A_{ij} = A_{ij}^{old}$, which does not introduce the $1/\rho_s(0)$ factor that appears in the FDS 6.7.7 equation (from $Y^{n_s} = \left(\rho_s / \rho_s(0)\right)^{n_s}$, with $n_s=1$). Therefore, I suspect I am overlooking something trivial.

I was not aware that the kinetic constants could not simply be used in the newer version, thank you for pointing that out. I will look into the TGA analysis.

Thanks a lot for your help and for replying quickly!

**Comment by mcgratta** [comment]
$r=A ( \rho/\rho_0)^n   \exp(-E/RT)$

$r \rho_0 = A (\rho/\rho_0^{n-1} )^n   \exp(-E/RT)$

$r' = (A/\rho_0^{n-1} ) \rho^n  \exp(-E/RT)$

$r' = A' \rho^n  \exp(-E/RT)$

**Reply by r-broek** [reply]
Ah I see! Thanks for your patience, it is much appreciated.

**Comment by mcgratta** [comment]
r-broek wrote:

In the FDS user guide (p.93), the footnote mentions that the theory of the
 kinetic parameter estimation holds only for $n_S = 1$. The general theory
 given in the verification guide (p. 193) starts with the equation:
 $\frac{d Y}{dT} = - A Y \exp{(-E / R T)}$.
 This equation indeed assumes $n_S = 1$, but does it not assume also that $n_T
 = 0$ and $n_{O_2} = 0$? I am not so concerned with the $n_T = 0$, since
 the estimated $A$ and $E$ predict the TGA curve reasonably well for water
 vapour. However, I would like to derive estimates for the kinetic
 parameters of the char oxidation process, for which if I understand
 correctly $n_{O_2} = 1$ (from eq. 17.8, user guide). Could you give me a
 hint on how to proceed?

I don't know how to do this.

**Comment by mcgratta** [comment]
I do not know how to derive kinetic parameters for char oxidation.

**Reply by r-broek** [reply]
Thank you for the reply. After posting, I realized there was still something I could try, so I deleted the post. I apologize for any inconvenience this may have caused.

While estimating the kinetic parameters based on the general theory may not be fully accurate for reactions that are not first-order, using PYROLYSIS_RANGE and REFERENCE_TEMPERATURE for the char oxidation reaction still seems to work well.

**Comment by mcgratta** [comment]
Are you assuming that oxygen is involved in the reaction?

**Reply by r-broek** [reply]
Yes, I am using the SURFACE_OXIDATION_MODEL.
